core/services/gateway/handlers/functions/orm.go

```diff
@@ -130,3 +134,37 @@ func (cs *cachedSubscriptionRow) encode() CachedSubscription {
 		},
 	}
 }
+
+func (o *orm) GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error) {
+	var address []common.Address
+	stmt := fmt.Sprintf(`
+		SELECT allowed_address
+		FROM %s
+		WHERE router_contract_address = $1
+		ORDER BY id ASC
+		OFFSET $2
+		LIMIT $3;
+	`, allowlistTableName)
+	err := o.q.WithOpts(qopts...).Select(&address, stmt, o.routerContractAddress, offset, limit)
+	if err != nil {
+		return address, err
+	}
+
+	return address, nil
+}
+
+func (o *orm) UpsertAllowedSender(id uint64, allowedSender common.Address, qopts ...pg.QOpt) error {
+	stmt := fmt.Sprintf(`
+		INSERT INTO %s (id, allowed_address, router_contract_address)
+		VALUES ($1,$2,$3) ON CONFLICT (id, router_contract_address) DO UPDATE
+		SET allowed_address=$2, router_contract_address=$3;`, allowlistTableName)
+
+	_, err := o.q.WithOpts(qopts...).Exec(
+		stmt,
+		id,
+		allowedSender,
+		o.routerContractAddress,
+	)
+
```

Comment:
Can we add a debug log message here indicating that we successfully updated the stored allowlist in the DB? Include the length too.

Reply:
not sure what do you mean by length, rows affected? this will only store one, wdyt about logging the sender address and router contract address?

Reply:
I think the more logs the better. It will make debugging much easier if we ever encounter bugs.

Reply:
The log looks good now. Thanks!